extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -348,15 +348,14 @@ private String signJwtWithClientSecret(OidcTenantConfig cfg) {
     }
 
     private void processSuccessfulAuthentication(RoutingContext context, TenantConfigContext configContext,
-            AccessToken result, SecurityIdentity securityIdentity) {
-
+            AccessToken result, String refreshToken, SecurityIdentity securityIdentity) {
         removeCookie(context, configContext, getSessionCookieName(configContext));
 
-        String cookieValue = new StringBuilder(result.opaqueIdToken())
-                .append(COOKIE_DELIM)
-                .append(result.opaqueAccessToken())
-                .append(COOKIE_DELIM)
-                .append(result.opaqueRefreshToken()).toString();
+        String refresh = result.opaqueRefreshToken() != null ? result.opaqueRefreshToken() : refreshToken;
```

Comment:
This line can be removed once the proposed changes have been implemented

Reply:
You mean, rather `result.opaqueRefreshToken()` being replaced with `refreshToken` (if that's always set from either two at the method invocation already, as you proposed, right?

Reply:
@sdaschner, this method is called from 2 places, when we complete the code grant exchange and refresh the token, in the former case RT is expected to be avail, in the latter - may not be, so I thought that in this function we should use `refreshToken` parameter

Reply:
Yes, exactly +1